torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1471,13 +1471,29 @@ void ComputeConstant(Node* n, int opset_version) {
         if (input0_shape_size.has_value()) {
           auto input0_shape_value = input0_shape_size.value();
           if (ConstantValueMap::HasValue(n->input(1)->debugName())) {
+            // When value of `shape` is statically known.
+            // Shape of expand output can be computed.
             auto shape_temp = ConstantValueMap::GetValueInto1DInt64Vector(
                 n->input(1)->debugName());
             auto final_shape =
                 ComputeShapeFromExpand(input0_shape_value, shape_temp);
             if (final_shape.has_value()) {
               UpdateShape(n->output(), final_shape.value());
             }
+          } else if (ConstantValueMap::HasShape(n->input(1)->debugName())) {
```

Comment:
`GetShapeInto1DInt64VectorWithOneUnknown` returns `nullopt` if `HasShape` returns false, so I think it's redundant to call `HasShape` here.